app/services/technical_metadata_generator.rb

```diff
@@ -2,8 +2,8 @@
 
 # Generates and persists technical metadata.
 class TechnicalMetadataGenerator
-  def self.generate(druid:, filepaths:, force: false)
-    new(druid: druid, force: force).generate(filepaths)
+  def self.generate(druid:, filepath_map:, force: false)
```

Comment:
Now uses the provided filenames instead of generating the filename from filepath within this service. This allows for different strategies for deriving the filename depending on where this is called from.